scaper/core.py

```diff
@@ -358,6 +359,78 @@ def _validate_distribution(dist_tuple):
                 'number that is equal to or greater than trunc_min.')
 
 
+def _ensure_valid_source_time_tuple_bounds(source_time, source_duration, event_duration):
+    '''
+    Modify a source_time distribution tuple according to the duration of the
+    source and the duration of the event. This allows you to sample from 
+    anywhere in a source file without knowing the exact duration of every
+    source file. 
+
+    Parameters
+    ----------
+    source_time : tuple
+        Tuple specifying a distribution to sample from. See Scaper.add_event
+        for details about the expected format of the tuple and allowed values.
+    source_duration : float
+        Duration of the source audio file.
+    event_duration : float
+        Duration of the event to be extracted from the source file.
+
+    See Also
+    --------
+    Scaper.add_event : Add a foreground sound event to the foreground
+    specification.
+    '''
+    _validate_distribution(source_time)
+    old_source_time = deepcopy(source_time)
+    source_time = list(source_time)
+    eps = 1e-1 # epsilon to take care of when low == high.
```

Comment:
How should this be done? Should we change the distribution to be `const`? Or add some logic into the `_sample_uniform`, `_sample_trunc_norm` etc functions?

Reply:
I changed the distribution to `const` in the current implementation. It doesn't break anything and it seemed like the nicest solution so that sampling source_time is always happening in a similar way.